src/openforms/prefill/api/serializers.py

```diff
@@ -17,6 +24,46 @@ class PrefillPluginSerializer(PluginBaseSerializer):
     )
 
 
+PrefillPluginPredicate = Callable[[BasePlugin], bool]
+
+
+class PrefillPluginPredicateSerializer(serializers.Serializer[PrefillPluginPredicate]):
+    component_type = serializers.CharField(
+        required=False,
+        label=_("Form.io component type"),
+        help_text=_("Only return plugins applicable for the specified component type."),
+    )
+
+    def to_internal_value(self, data):
+        return super().to_internal_value(underscoreize(data))
```

Comment:
this bit would be worth breaking out into `openforms.api.serializers.FilterSerializer` as a base class, with a big fat disclaimer/pointer to #611

Then the existing query parameter serializers can be updated to subclass this base.

Reply:
Breaking this out separates the "underscorizing" in this method from the "camelcasing" in `as_openapi_params` and then *that* might be forgotten.

Keeping these 2 lines of together makes it clearer what's going on without having to inspect the whole mro.